argo/providers/vai/provider.go

```diff
@@ -316,11 +316,16 @@ func (vaip VAIProvider) EnqueueRun(ctx context.Context, providerConfig VAIProvid
 
 	var runId string
 
-	if runIntent.RunName.Name != "" {
+	if !runIntent.RunName.Empty() {
 		runId = fmt.Sprintf(runIntent.RunName.Name)
-	} else if runIntent.RunConfigurationName != "" {
-		runId = fmt.Sprintf("%s-%s", runIntent.RunConfigurationName, uuid.New().String())
-		runLabels[labels.RunConfiguration] = runIntent.RunConfigurationName
+	} else if !runIntent.RunConfigurationName.Empty() {
+		runId = fmt.Sprintf("%s-%s", runIntent.RunConfigurationName.Name, uuid.New().String())
```

Comment:
not using namespaced name as the provider id as this would be inconsistent with how we treat runs. I think this should be a separate change.